[pt-br] Localize content/pt/docs/concepts/components.md (#5396)
open-telemetry/opentelemetry.io: content/pt/docs/concepts/components.md

---
title: Componentes
description: Os principais componentes que compõem o OpenTelemetry

> aliases: [coleta de dados]

Reviewer: ~~~suggestion
~~~

weight: 20
default_lang_commit: 2f34c456ab38b4d3502cd07bc36fa1455d4ef875
---

O OpenTelemetry é atualmente composto por vários componentes principais:

- [Especificação](#specification)
- [Collector](#collector)
- [Implementações de API e SDK específicas](#language-specific-api--sdk-implementations)
  - [Bibliotecas de Instrumentação](#instrumentation-libraries)
  - [Exporters](#exporters)
  - [Instrumentação sem código](#zero-code-instrumentation)
  - [Detectores de Recursos](#resource-detectors)
  - [Propagadores entre Serviços](#cross-service-propagators)
  - [Amostragens](#samplers)

> - [Kubernetes Operator](#kubernetes-operator)

Reviewer 2: ```suggestion
- [Kubernetes operator](#kubernetes-operator)
```

- [Função como Serviço](#function-as-a-service-assets)

O OpenTelemetry permite que você substitua o uso de SDKs ou ferramentas
específicas para gerar e exportar dados de telemetria.

## Especificação {#specification}

> "Descreve os requisitos e expectativas entre diferentes linguagens para todas as

Reviewer 2: ```suggestion
Descreve os requisitos e expectativas entre diferentes linguagens para todas as
```

implementações. Além de uma definição de termos, a especificação define o
seguinte:

- **API:** Define tipos de dados e operações para gerar e correlacionar dados de
  rastreamento, métricas e logs.
- **SDK:** Define os requisitos para implementação específica de linguagem da
  API. São definidos aqui também os conceitos de configuração, processamento e
  exportação de dados.
- **Data:** Define o Protocolo OpenTelemetry (OTLP) e as convenções semânticas
  independentes de fornecedores que um backend de telemetria pode suportar.

Para mais informações, consulte [especificação](/docs/specs/).

## Collector

> O Coletor é um proxy para qualquer sistema de telemetria que pode receber...,

Reviewer 2: ```suggestion
O Colletor é um proxy para qualquer sistema de telemetria que pode receber...,
```

> O Colletor é um proxy para qualquer sistema de telemetria que pode receber...,

Reviewer 2: ```suggestion
O Collector é um proxy para qualquer sistema de telemetria que pode receber...,
```

processar e exportar dados de telemetria. Ele suporta o recebimento de dados de
telemetria em vários formatos. (Por exemplo, OTLP, Jaeger, Prometheus, bem como
muitas outras ferramentas ) .e enviar dados para um ou mais sistema de
telemetria. Ele também suporta o processamento e a filtragem de dados de
telemetria antes de serem exportados.

Para mais informações, consulte [Coletor](/docs/collector/).

## Implementações de API e SDK específicas {#language-specific-api--sdk-implementations}

O OpenTelemetry também possui SDKs para as linguagens mais populares do mercado
e permitem usar a API do OpenTelemetry para gerar dados de telemetria e exportar
esses dados para qualquer sistema de telemetria. Esses SDKs também permitem que
você utilize qualquer biblioteca e ou frameworks dessas linguagens da
instrumentação manual.

Para mais informações, consulte
[Instrumentando](/docs/concepts/instrumentation/).

### Bibliotecas de instrumentação {#instrumentation-libraries}

O OpenTelemetry suporta um grande número de componentes que geram dados de
telemetria relevantes a partir de bibliotecas e frameworks populares para as

> linguagens suportadas. Por exemplo, Solicitações HTTP de entrada e saída de uma

Reviewer 2: ```suggestion
linguagens suportadas. Por exemplo, requisições HTTP de entrada e saída de uma
```

> biblioteca HTTP geram dados sobre essas solicitações. Um objetivo Aspiracional

Reviewer 2: ```suggestion
biblioteca HTTP geram dados sobre essas requisições. Um objetivo Aspiracional
```

Para mais informações, consulte
[Bibliotecas de instrumentação](/docs/concepts/instrumentation/libraries/).

### Exporters

{{% docs/languages/exporters/intro %}}

### Instrumentação sem código {#zero-code-instrumentation}

OpenTelemetry possibilita essa forma de instrumentar sua aplicação sem alterar
seu código-fonte. Embora o mecanismo dependa da linguagem, a instrumentação sem
código-fonte adiciona as capacidades da API e do SDK do OpenTelemetry à sua
aplicação. Além disso, pode adicionar um conjunto de bibliotecas de
instrumentação e dependências do exportador.

Para mais informações, consulte
[Instrumentação sem código](/docs/concepts/instrumentation/zero-code/).

### Detectores de recursos {#resource-detectors}

Um [recurso](/docs/concepts/resources/) representa a entidade que produz
telemetria como atributos de recurso. Por exemplo, um processo que produz
telemetria que está sendo executado em um contêiner no Kubernetes tem um nome de
Pod, um namespace e possivelmente um nome de implantação. Você pode incluir
todos esses atributos no recurso. As implementações específicas de linguagem do
OpenTelemetry fornecem detecção de recursos a partir da variável de ambiente

> OTEL_RESOURCE_ATTRIBUTES e para muitas entidades comuns, como tempo de execução

Reviewer 2: ```suggestion
`OTEL_RESOURCE_ATTRIBUTES` e para muitas entidades comuns, como tempo de execução
```

---
title: Componentes
description: Os principais componentes que compõem o OpenTelemetry
aliases: [coleta de dados]
weight: 20
default_lang_commit: 3386fa71833d0c25dac410a5db66ae93dc61afcb
---

O OpenTelemetry é atualmente composto por vários componentes principais:

- [Especificação](#especificação)

> - [Coletor](#coletor)

Reviewer 3: ```suggestion
- [Collector](#coletor)
```
Como se trata de um componente do projeto, acordamos em não traduzir.

Reviewer 3: https://docs.google.com/document/d/1kyu6HgdsM3-iDyf3OuRmfw4SwNJy9BVui6nn5JsKb5I/edit?tab=t.0#heading=h.o389kqcsmupl

Author: certo !

- [Implementações de API e SDK específicas](#implementações-de-api--sdk-específicas)
  - [Bibliotecas de Instrumentação](#bibliotecas-de-instrumentação)

>   - [Exportadores](#exportadores)

Reviewer 3: ```suggestion
  - [Exporters](#exporters)
```
Como se trata de um componente do projeto, acordamos em não traduzir.

Reviewer 3: https://docs.google.com/document/d/1kyu6HgdsM3-iDyf3OuRmfw4SwNJy9BVui6nn5JsKb5I/edit?tab=t.0#heading=h.o389kqcsmupl

Author: ok !

>   - [Auto-Instrumentação](#auto-instrumentação)

Reviewer 3: ```suggestion
  - [Auto-Instrumentação](#auto-instrumentação)
```
O termo Auto-instrumentation foi substituído por Zero-code Instrumentation, após uma conversa definimos que a tradução fica: Instrumentação sem código. 

https://docs.google.com/document/d/1kyu6HgdsM3-iDyf3OuRmfw4SwNJy9BVui6nn5JsKb5I/edit?tab=t.0#heading=h.o389kqcsmupl

Author: certo

Author: Alteração feita

  - [Detectores de Recursos](#resource-detectors)
  - [Propagadores entre Serviços](#cross-service-propagators)

>   - [Amostragen](#samplers)

Reviewer 3: ```suggestion
  - [Amostragem](#samplers)
```

> - [Operador Kubernetes](#operador-kubernetes)

Reviewer 3: ```suggestion
- [Operator Kubernetes](#operador-kubernetes)
```
Não tenho certeza, acredito que manter o termo operator fica melhor. Operator k8s é comum na area tech.

Author: certo

> - [Função como Serviço](#função-como-serviço)

Reviewer: please do not translate anchors, see https://opentelemetry.io/docs/contributing/localization/ for more details

Reviewer 3: @Lindemberg-Chagas aqui explicação sobe ancoras:
https://github.com/edsoncelio/otel-localization-ptbr-guide/blob/main/README.md
![image](https://github.com/user-attachments/assets/c987f42d-9c0b-4a12-9841-666ac3d5aaa7)

Author: OK, não sabia o que era esse anchor, vou ver isso

Author: Alteração Feita

O OpenTelemetry permite que você substitua o uso de SDKs ou ferramentas específicas para gerar e exportar dados de telemetria.

> ## Especificação

Reviewer 3: ```suggestion
## Especificação {#specification}
```

Tem um github actions quebrando devido a falta das ancoras.

"Descreve os requisitos e expectativas entre diferentes linguagens para todas as implementações. Além de uma definição de termos, a especificação define o seguinte:

> - **API:** Define os tipos dos dados e das operações que são feitas para correlacionar com os dados de rastreamento, métricas e logs..

Reviewer 3: ```suggestion
- **API:** Define tipos de dados e operações para gerar e correlacionar dados de rastreamento, métricas e logs.
```

- **SDK:** Define os requisitos para implementação em uma linguagem especifica. O onceitos de configuração, processamento de dados e exportação também são definidos aqui.

> - **Data:** Define o Protocolo OpenTelemetry (OTLP) para qualquer sistema de telemetria, independentemente do fornecedor ou tecnologia específica que você está utilizando.

Reviewer 3: ```suggestion
- **Data:** Define o Protocolo OpenTelemetry (OTLP) e as convenções semânticas independentes de fornecedores que um backend de telemetria pode suportar.
```

Para mais informações, consulte [especificação](/docs/specs/).

> ## Coletor

Reviewer 3: ```suggestion
## Collector
```

O Coletor é um proxy para qualquer sistema de telemetria que pode receber...,
processar e exportar dados de telemetria. Ele suporta o recebimento de dados de telemetria em vários formatos. (Por exemplo, OTLP, Jaeger, Prometheus, bem como muitas outras ferramentas ) .e enviar dados para um ou mais sistema de telemetria. Ele também suporta o processamento e a filtragem de dados de telemetria antes de serem exportados.

Para mais informações, consulte [Coletor](/docs/collector/).

## Implementações de API e SDK específicas.

O OpenTelemetry também possui SDKs para as linguagens mais populares do mercado e permitem usar a API do OpenTelemetry para gerar dados de telemetria e exportar esses dados para qualquer sistema de telemetria. Esses SDKs também permitem que você utilize qualquer biblioteca e ou frameworks dessas linguagens da instrumentação manual.

Para mais informações, consulte [Instrumentando](/docs/concepts/instrumentation/).

> ### Bibliotecas de instrumentação

Reviewer 3: ```suggestion
### Bibliotecas de instrumentação {#instrumentation-libraries}
```

O OpenTelemetry suporta um grande número de componentes que geram dados de telemetria relevantes a partir de bibliotecas e frameworks populares para as linguagens suportadas. Por exemplo, Solicitações HTTP de entrada e saída de uma biblioteca HTTP geram dados sobre essas solicitações. Um objetivo Aspiracional do OpenTelemetry é que todas as bibliotecas populares sejam construídas para serem observáveis por padrão, de modo que dependências separadas não sejam necessárias.

Para mais informações, consulte
[Bibliotecas de instrumentação](/docs/concepts/instrumentation/libraries/).

> ### Exportadores

Reviewer 3: ```suggestion
### Exporters
```

Olhando nosso glossário exporter não foi traduzido se trata de um recurso do projeto.

{{% docs/languages/exporters/intro %}}

### Instrumentação sem código

OpenTelemetry possibilita essa forma de instrumentar sua aplicação sem alterar seu código-fonte. Embora o mecanismo dependa da linguagem, a instrumentação sem código-fonte adiciona as capacidades da API e do SDK do OpenTelemetry à sua aplicação. Além disso, pode adicionar um conjunto de bibliotecas de instrumentação e dependências do exportador.

Para mais informações, consulte
[Instrumentação sem código](/docs/concepts/instrumentation/zero-code/).

> ### Detectores de recursos

Reviewer 3: ```suggestion
### Detectores de recursos {#resource-detectors}
```

Um [recurso](/docs/concepts/resources/) representa a entidade que produz telemetria como atributos de recurso. Por exemplo, um processo que produz telemetria que está sendo executado em um contêiner no Kubernetes tem um nome de Pod, um namespace e possivelmente um nome de implantação. Você pode incluir todos esses atributos no recurso. As implementações específicas de linguagem do OpenTelemetry fornecem detecção de recursos a partir da variável de ambiente OTEL_RESOURCE_ATTRIBUTES e para muitas entidades comuns, como tempo de execução do processo, serviço, host ou sistema operacional.

Para mais informações, consulte [Recursos](/docs/concepts/resources/).

> ### Propagadores entre serviços

Reviewer 3: ```suggestion
### Propagadores entre serviços {#cross-service-propagators}
```

A propagação é o mecanismo que move dados entre serviços e processos.
Embora não se limite ao rastreamento, a propagação permite que os rastros construam informações sobre um sistema através de serviços que estão distribuídos arbitrariamente entre limites de processos e redes.

> Na grande maioria dos casos de uso, a propagação de contexto ocorre por meio de bibliotecas de instrumentação. Se for necessário, você pode usar propagadores manualmente para serializar e desserializar preocupações transversais, como o contexto de um span. [metadados](/docs/concepts/signals/baggage/).

Reviewer 3: ```suggestion
Na grande maioria dos casos de uso, a propagação de contexto ocorre por meio de bibliotecas de instrumentação. Se for necessário, você pode usar propagadores manualmente para serializar e desserializar aspectos como o contexto de um trecho e a [bagagem](/docs/concepts/signals/baggage/).
```

Achei um pouco confuso o trecho `preocupações transversais`, essa é uma recomendação que pode ficar melhor.

> ### Amostragem

Reviewer 3: ```suggestion
### Amostragem {#samplers}
```

> A amostragem é um processo que restringe a quantidade de rastros que são gerados por um sistema. OpenTelemetry oferece várias formas de fazer essa para linguagens mais populares do mercado.

Reviewer 3: ```suggestion
A amostragem é um processo que restringe a quantidade de rastros que são gerados por um sistema. Cada implementação específica de linguagem do OpenTelemetry oferece [amostradores pela cabeça](/docs/concepts/sampling/#head-sampling).
```

A página sampling foi traduzida, head samplers foi traduzido como amostragem pela cabeça.
https://opentelemetry.io/pt/docs/concepts/sampling/

Ainda não estou satisfeito com essa frase, no momento não tenho muita ideia para melhorar o contexto.

[Amostragem](/docs/concepts/sampling/#head-sampling).

Para mais informações, consulte [Amostragem](/docs/concepts/sampling).

> ## Operador Kubernetes

Reviewer 3: ```suggestion
## Operador Kubernetes {#kubernetes-operator}
```

> O OpenTelemetry Operator é a implementação de um Operador Kubernetes. O operador gerencia o OpenTelemetry Collector e a auto-instrumentação das cargas de trabalho usando OpenTelemetry.

Reviewer 3: ```suggestion
O OpenTelemetry Operator é a implementação de um Operador Kubernetes. O operador gerencia o OpenTelemetry Collector e a instrumentação sem código das cargas de trabalho que utilizam OpenTelemetry.
```

Para mais informações, consulte [Kubernetes Operator](/docs/kubernetes/operator/).

> ## Função como Serviço

Reviewer 3: ```suggestion
## Função como Serviço {#function-as-a-service-assets}
```

> O OpenTelemetry suporta vários métodos de monitoramento da Função como Serviço fornecidos por diferentes sistema de telemetria. A comunidade do OpenTelemetry atualmente fornece camadas Lambda pré-construídas capazes de auto-instrumentar sua aplicação, bem como a opção de uma camada Lambda de Collector independente que pode ser usada ao instrumentar aplicações manualmente ou automaticamente.

Reviewer 3: ```suggestion
O OpenTelemetry oferece vários métodos de monitoramento de Function-as-a-Service fornecidos por diferentes provedores de nuvem. A comunidade OpenTelemetry atualmente disponibiliza camadas Lambda pré-construídas, capazes de realizar instrumentação sem código na sua aplicação, bem como a opção de uma camada Lambda do Coletor independente, que pode ser usada ao instrumentar aplicações manualmente ou automaticamente.
```
